src/Authentication/Authentication/Helpers/AuthenticationHelpers.cs

```diff
@@ -71,6 +75,27 @@ internal static IAuthenticationProvider GetAuthProvider(IAuthContext authContext
             }
             return authProvider;
         }
+        /// <summary>
+        ///     Gets a certificate based on the current context.
+        /// </summary>
+        /// <param name="context">Current <see cref="IAuthContext"/> context</param>
+        /// <returns>A <see cref="X509Certificate2"/> based on provided <see cref="IAuthContext"/> context</returns>
+        private static X509Certificate2 GetCertificate(IAuthContext context)
+        {
+            if (!string.IsNullOrWhiteSpace(context.CertificateThumbprint))
+            {
+                return GetCertificateByThumbprint(context.CertificateThumbprint);
+            }
+            
+            else if (!string.IsNullOrWhiteSpace(context.CertificateName))
+            {
+                return GetCertificateByName(context.CertificateName);
+            }
+            else
+            {
+                return context.Certificate;
```

Comment:
if a certificate does exist shouldn't it take precedence over thumprint or name

Reply:
The original two certificate acquisition methods (by Name and Thumbprint) should take precedence since they are what our customers were used to, It would be the least surprising change.